woo-razorpay.php

```diff
@@ -2274,21 +2274,53 @@ public function updateGiftAndCoupon($razorpayData, $order, $orderId, $razorpayPa
 
                             if($balance == null && $balance >= 0 && $usedAmt > $balance ){
                                 // initiate refund in case gift card faliure
-                                $this->processGiftCardRefund($orderId, $razorpayPaymentId, $razorpayData['amount_paid'], $reason = '');
+                                $this->processRefund($orderId, $razorpayPaymentId, $razorpayData['amount_paid'], $reason = '');
                             }else{
                                 //Deduct amount of gift card
                                $this->debitGiftCards($orderId, $order, "order_id: $orderId checkout_update_order_meta", $usedAmt, $giftCode);
                             }
 
                         }else{
-                            $this->processGiftCardRefund($orderId, $razorpayPaymentId, $razorpayData['amount_paid'], $reason = '');
+                            $this->processRefund($orderId, $razorpayPaymentId, $razorpayData['amount_paid'], $reason = '');
                         }
 
                     }else{
-                       $this->processGiftCardRefund($orderId,  $razorpayPaymentId, $razorpayData['amount_paid'], $reason = '');
+                       $this->processRefund($orderId,  $razorpayPaymentId, $razorpayData['amount_paid'], $reason = '');
                     }
 
-                }else{
+                }else if($promotion['type'] == 'terra_wallet') {
+                    if (!is_plugin_active('woo-wallet/woo-wallet.php')) {
+                        $this->processRefund(
+                            $orderId,
+                            $razorpayPaymentId,
+                            $razorpayData['amount_paid'],
+                            'Terra Wallet Plugin is not active');
+                    }
+                    $viaWalletAmount = $promotion['value']/100;
+                    $walletBalance = woo_wallet()->wallet->get_wallet_balance($order->get_user_id(), 'edit');
+                    if ($walletBalance < $viaWalletAmount)
```

Comment:
Need to understand the logic.

Reply:
if the walletbalance(amount available in terrawallet) is less than viaWalletAmount(amount redeemed for an order) then i am processing refund as there is insufficient balance